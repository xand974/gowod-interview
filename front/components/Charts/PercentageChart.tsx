import React from "react";
import { StyleSheet, Text, View } from "react-native";
import styled from "styled-components/native";
import { WHITE } from "assets/styles/@core.style";

const PercentageView = styled.View`
  border-color: ${WHITE};
  padding: 10px;
  border-width: 4px;
  border-radius: 100px;
  width: 100px;
  height: 100px;
  align-items: center;
  justify-content: center;
  margin: 10px 0;
`;

const PercentageText = styled.Text`
  color: ${WHITE};
  font-size: 20px;
  font-weight: bold;
`;

interface PercentageChartProp {
  total: number;
}
export default function PercentageChart({ total }: PercentageChartProp) {
  return (
    <PercentageView>
      <PercentageText>{total?.toFixed()}%</PercentageText>
    </PercentageView>
  );
}

const styles = StyleSheet.create({});
