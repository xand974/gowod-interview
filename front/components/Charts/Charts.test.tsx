import { render, screen } from "@testing-library/react";

import renderer from "react-test-renderer";
import PercentageChart from "./PercentageChart";

test("renders react component", () => {
  const tree: any = renderer.create(<PercentageChart total={10} />).toJSON();
  console.log(tree);

  expect(tree?.children?.length).toBe(1);
  expect(2 + 3).toEqual(5);
});
