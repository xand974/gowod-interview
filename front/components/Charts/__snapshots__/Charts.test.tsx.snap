// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render bars chart 1`] = `
<View
  detailed={false}
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "center",
        "marginTop": 30,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      detailed={false}
      id={0}
      item={
        Object {
          "name": "ankles",
          "percentage": 10,
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#98ffcf",
            "height": "1000%",
            "left": 10,
            "marginRight": 10,
            "position": "relative",
            "width": 3,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      detailed={false}
      id={1}
      item={
        Object {
          "name": "shoulders",
          "percentage": 20,
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#98ffcf",
            "height": "667%",
            "left": 10,
            "marginRight": 11,
            "position": "relative",
            "width": 3,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`render percentage chart 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderRadius": 100,
        "borderWidth": 4,
        "height": 100,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": 100,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    10
    %
  </Text>
</View>
`;
